spikeinterface_gui/unitlist.py

```diff
@@ -530,16 +600,53 @@ def _panel_get_sorted_indices(self):
             new_indices = [original_unit_ids.index(unit_id) for unit_id in sorted_unit_ids]
             return new_indices
 
+    def _panel_on_hide_noise(self, event):
+        if self.hide_noise.value:
+            # Filter out noise units
+            self.df = self.df.query("quality != 'noise'")
+        else:
+            # Show all units
+            self.df = self.df_full
+            labels = [self.controller.get_unit_label(unit_id, category="quality") for unit_id in self.controller.unit_ids]
+            self.df["quality"] = labels
+        self.controller.filtered_unit_ids = self.df.index.values
+        self.table.value = self.df
+        self.table.selection = []
+        self.refresh()
+        self._panel_refresh_header()
+
+    def _panel_on_show_only(self, event):
+        if self.label_definitions is not None and "quality" in self.label_definitions:
+            value = event.new
+            if value == "all":
+                # Show all units
+                self.df = self.df_full
+                labels = [self.controller.get_unit_label(unit_id, category="quality") for unit_id in self.controller.unit_ids]
+                self.df["quality"] = labels
+            else:
+                # Filter out units that don't match the selected label
+                self.df = self.df.query(f"quality == '{value}'")
+            self.controller.filtered_unit_ids = self.df.index.values
+            self.table.value = self.df
+            self.table.selection = []
+            self.refresh()
+            self._panel_refresh_header()
+            self.hide_noise.value = False
+
     def _panel_handle_shortcut(self, event):
+        if event.data in ["delete", "merge", "good", "mua", "noise"]:
+            if not self.controller.curation:
+                return
         if event.data == "delete":
             self.delete_unit()
         elif event.data == "merge":
-            self.merge_selected()
+            if self.controller.curation:
+                self.merge_selected()
         elif event.data == "visible":
             selected_rows = self._panel_get_selected_unit_ids()
             for unit_id in self.controller.unit_ids:
                 self.controller.unit_visible_dict[unit_id] = False
-            for unit_id in self.controller.unit_ids[selected_rows]:
+            for unit_id in self.df.index.values[selected_rows]:
```

Comment:
why ?

Reply:
I think that the controller should always control the internal state instead of delegating states to the veiws.